utils/src/main/scala/za/co/absa/enceladus/utils/udf/UDFLibrary.scala

```diff
@@ -27,60 +27,76 @@ import scala.collection.mutable
 import scala.util.{Failure, Success, Try}
 
 class UDFLibrary()(implicit val spark: SparkSession) {
+  UDFLibrary.registerUdf() // re-registration would be unnecessary
+}
 
-  spark.udf.register(stdCastErr, { (errCol: String, rawValue: String) =>
-    ErrorMessage.stdCastErr(errCol, rawValue)
-  })
+object UDFLibrary {
 
-  spark.udf.register(stdNullErr, { errCol: String => ErrorMessage.stdNullErr(errCol) })
+  private var isRegistered = false
```

Comment:
I know it's a small use case, but could there be a way to avoid using this var?

Reply:
I thought about leveraging `lazy` or some AtomicReference, but their logic is similar. It seems that for Spark3 the UDF registration works differently and thus, Spark3 is complaining about reregistering these UDFs. But there are still some leftovers. I will add a point to address it in the UDF task, it is related anyway.